frontend/src/app/libs/common-components/components/select/select.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, Input, ChangeDetectionStrategy, forwardRef, Output, EventEmitter, OnInit, Optional, Host, SkipSelf, ElementRef } from '@angular/core';
+import { NG_VALUE_ACCESSOR, FormControl, ControlContainer, FormGroup } from '@angular/forms';
+
+@Component({
+	selector: 'app-select',
+	templateUrl: './select.component.html',
+	styleUrls: ['./select.component.scss'],
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: forwardRef(() => SelectComponent),
+			multi: true
+		}
+	],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SelectComponent implements OnInit {
+	public icon: string = 'keyboard_arrow_down';
+	public opened: boolean = false;
+	@Input() public label: string;
+	@Input() public formControlName: string;
+	@Input() public formControl: FormControl;
+	@Input() public value: number | string | null | undefined;
+	@Output() public valueChange: EventEmitter<number | string | boolean | null | undefined> = new EventEmitter();
+
+	public constructor(
+		@Optional() @Host() @SkipSelf() protected parentFormContainer: ControlContainer,
+		public elementRef: ElementRef
+	) { }
+
+	public ngOnInit(): void {
+		this.formControlName = this.elementRef.nativeElement.getAttribute('formControlName');
```

Comment:
And this component the same.
So, I suppose we need base component for all these.